content/smyja/how-to-use-graphql-with-django.md

```diff
@@ -0,0 +1,296 @@
+    restaurants = graphene.List(RestaurantType)
+
+    def resolve_restaurants(self, info, **kwargs):
+      return Restaurant.objects.all()
+
+```
+Start the django server with `python manage.py runsrver` then visit the `/graphql` route to see the api browser, it should look like this
+
+![Api browser](https://raw.githubusercontent.com/Smyja/ugc/grahql-with-django/content/smyja/api-browser.png)
+
+
+To get the list of restaurants with specific data like name and address, run a query with this:
+
+```graphql
+query {
+    restaurants {
+        id
+        name
+        address
+    }
+}
+
+```
+
+The output should look like this:
+
+![list of restaurants](https://raw.githubusercontent.com/Smyja/ugc/grahql-with-django/content/smyja/restaurant-list.png)
+
+To modify any data in our database we would need to create a mutation.
+Below is the `CreateRestaurant` mutation, which we will add to the `schema.py` file.
+
+```py
+class CreateRestaurant(graphene.Mutation):
+    class Arguments:
+        name = graphene.String()
+        address = graphene.String()
+
+    ok = graphene.Boolean() 
+    restaurant = graphene.Field(RestaurantType)
+
+    def mutate(self, info, name, address):
+        restaurant = Restaurant(name=name, address=address)
+        restaurant.save()
+
+        return CreateRestaurant(ok=True, restaurant=restaurant)
+```
+
+The `CreateRestaurant` mutation takes in the model fields as arguments while the `mutate()` function is where the db change happens using Django's object-relational mapper (ORM).
+
+Create a `Mutation` class then initialize the mutation with the schema.
+
+```py
+class Mutation(graphene.ObjectType):
+    create_restaurant = CreateRestaurant.Field()
+```
+
+After adding the mutation and query, define the schema at the end of the `schema.py` file.
+
+```py
+schema = graphene.Schema(query=Query, mutation=Mutation)
+```
+
+Start the server and a run a mutation with the GraphQL API browser using this:
+
+```graphql
+mutation {
+    createRestaurant(name: "Kada Plaza", address: "Lekki GARDENS") {
+        ok
+        restaurant {
+            id
+            name
+            address
+        }
+    }
+}
+```
+
+The mutation returns a restaurant object with the fields that were passed in.
+
+To delete a restaurant, we would need to create a mutation. Below is the `DeleteRestaurant` mutation. Let's add it to our `schema.py` file.
+
+```python
+class DeleteRestaurant(graphene.Mutation):
+    class Arguments:
+        id = graphene.Int()
+
+    ok = graphene.Boolean()
+
+    def mutate(self, info, id):
+        restaurant = Restaurant.objects.get(id=id)
+        restaurant.delete()
+
+        return DeleteRestaurant(ok=True)
+```
+
+Next, we'll add the `DeleteRestaurant` mutation to the `Mutation` class:
+
+```py
+delete_restaurant = DeleteRestaurant.Field()
+```
+
+Run the mutation with the GraphQL API browser using this.
+
+```graphql  
+    mutation {
+        deleteRestaurant(id: 1) {
+            ok
+        }
+    }
```

Comment:
Is this `mutation` supposed to be nested inside some outer object-like structure? If not I suggest doing away with the leading whitespace:

```suggestion
mutation {
  deleteRestaurant(id: 1) {
    ok
  }
}
```